File: system-test/typescript/tests/legacy/mod11/contract/tax/BaseTaxContractTest.ts
import type { ContractTest } from '../base/BaseExternalSystemContractTest.js';

export function registerTaxContractTests(test: ContractTest): void {
  test('shouldBeAbleToGetTaxRate', async ({ scenario }) => {
    await scenario
      .given()
      .country()
      .withCode('US')
      .withTaxRate(0.09)
      .then()
      .country('US')
      .hasTaxRateIsPositive();
  });
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/legacy/mod11/contract/tax/BaseTaxContractTest.java
package com.optivem.shop.systemtest.legacy.mod11.contract.tax;

import com.optivem.shop.systemtest.legacy.mod11.contract.base.BaseExternalSystemContractTest;
import org.junit.jupiter.api.Test;

public abstract class BaseTaxContractTest extends BaseExternalSystemContractTest {
    @Test
    void shouldBeAbleToGetTaxRate() {
        scenario
                .given().country().withCode("US").withTaxRate(0.09)
                .then().country("US").hasTaxRateIsPositive();
    }
}
